pkg/lunii/writer.go

```diff
@@ -132,9 +133,31 @@ func (device *Device) AddStudioPack(studioPack *StudioPack, updateChan *chan str
 	deviceAudioDirectory := filepath.Join(tempPath, "sf", "000")
 	os.MkdirAll(deviceAudioDirectory, 0700)
 
+	// for each audioindex start a goroutine to convert and write the audio
+	// throttle the number of goroutines depending on the ram available
+
+	waitingTime := 500 * time.Millisecond // 500 milliseconds
+	scheduleMaxRetries := 20              // 10 seconds (20 * 500ms)
+	minMemory := uint64(500000000)        // 500MB
+
 	for i, audio := range *audioIndex {
+		// wait until there is enough memory to convert the audio
+		tryCount := 0
```

Comment:
My solution was much simpler:
	if i % 4  == 0 {
			wg.Wait()
	}
	
so that we only start up to 4 coroutines at a time.
And only consume something like 3GB of RAM on my computer and the pack we discussed.
The result is also slower than your proposal. By starting 8 coroutines I have roughly the same time than you, with a ~6 GB of RAM consumed.

Reply:
Yes indeed. You started 4 coroutines parallelly, then 6. I just made this number a dynamic variable so that it will use the most coroutine possible parallelly to speed up the process while still having enough memory for the system.

Great to know it works, it's probably not the most beautiful way, but i'll publish it and will leave the task of making it more elegant for later.